src/Microsoft.SymbolManifestGenerator/SymbolManifestGenerator.cs

```diff
@@ -0,0 +1,136 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.SymbolStore;
+using Microsoft.SymbolStore.KeyGenerators;
+
+namespace Microsoft.SymbolManifestGenerator
+{
+    public class SymbolManifestGenerator
+    {
+        private const int Private = 340;
+        public SymbolManifestGenerator()
+        {
+        }
+
+        public static void GenerateManifest(ITracer tracer, DirectoryInfo dir, string manifestFileName, DotNetPlatform platform = DotNetPlatform.NetFramework)
+        {
+            ManifestDataV1 manifestData = new();
+
+            foreach (FileInfo runtimeModule in GetRuntimeModules(dir, platform))
+            {
+                FileStream fileStream = runtimeModule.OpenRead();
+                SymbolStoreFile symbolStoreFile = new(fileStream, runtimeModule.FullName);
+                FileKeyGenerator generator = new(tracer, symbolStoreFile);
+                if (!generator.IsValid())
+                {
+                    tracer.Information($"Runtime module '{runtimeModule.FullName}' is not a valid file. Skipping.");
+                    continue;
+                }
+
+                IEnumerable<SymbolStoreKey> clrKeys = generator.GetKeys(KeyTypeFlags.ClrKeys);
+                foreach (SymbolStoreKey clrKey in clrKeys)
+                {
+                    FileInfo specialFile = GetSymbolFileToAddAdditionalDebugEntry(runtimeModule.DirectoryName, clrKey);
+                    if (specialFile == null)
+                    {
+                        tracer.Information($"Known special file '{clrKey.FullPathName}' for runtime module '{runtimeModule.FullName}' does not exist in directory '{runtimeModule.DirectoryName}'. Skipping.");
+                        continue;
+                    }
+
+                    string basedirRelativePath = Path.Combine(runtimeModule.DirectoryName.Replace(dir.FullName, string.Empty).TrimStart(Path.DirectorySeparatorChar), specialFile.Name);
+                    string fileHash = CalculateSHA512(specialFile);
+
+                    ManifestDataEntry manifestDataEntry = new()
+                    {
+                        basedirRelativePath = basedirRelativePath,
+                        symbolKey = clrKey.Index,
+                        sha512 = fileHash,
+                        debugInformationLevel = Private
+                    };
+
+                    manifestData.entries.Add(manifestDataEntry);
+                }
+            }
+
+            string manifestDataContent = JsonSerializer.Serialize(manifestData);
+            File.WriteAllText(manifestFileName, manifestDataContent);
+        }
+
+        private static IEnumerable<FileInfo> GetRuntimeModules(DirectoryInfo dir, DotNetPlatform platform)
+        {
+            string runtimeModule = platform switch
```

Comment:
This doesn't work for Linux (libcoreclr.so) and MacOS (libcoreclr.dylib) runtimes. I don't like that this knowledge is in more than one place (here and in the key generators).

Reply:
Ah, good catch. I mostly just wanted to minimize the number of times we grabbed ClrKeys, is there a way to quickly tell if a file is a runtime module? Should we have an API that does that?

I'll revert to just trying to grab ClrKeys for all files in `dir` like before